array.c

```diff
@@ -7536,25 +7536,21 @@ rb_ary_take(VALUE obj, VALUE n)
 
 /*
  *  call-seq:
- *    array.take_while {|element| ... } -> new_array
- *    array.take_while -> new_enumerator
- *
- *  Returns a new +Array+ containing zero or more leading elements of +self+;
- *  does not modify +self+.
+ *    take_while {|element| ... } -> new_array
+ *    take_while -> new_enumerator
  *
  *  With a block given, calls the block with each successive element of +self+;
- *  stops if the block returns +false+ or +nil+;
- *  returns a new +Array+ containing those elements for which the block returned a truthy value:
+ *  stops iterating if the block returns +false+ or +nil+;
+ *  returns a new array containing those elements for which the block returned a truthy value:
```

Comment:
Like `Array#take`, we should specify that the original array is not modified.

Reply:
Added.